### ISSUE TITLE
Use multiple standards at the same time

### ISSUE THREAD

**Issue by nbouhid** [ikappas/vscode-phpcs#21]
My team uses multiple standards like this:
`phpcs --standard=Drupal,DrupalSecure`
which works fine when running in on the command line

So we configured it as 

```
{
  "phpcs.standard": "Drupal,DrupalSecure"
}
```

When we do that, we get 
`phpcs: Unexpected token P in JSON at position 0`

Is there a way to achieve that?

**Comment by ikappas** [comment]
@nbouhid Thank you for reporting this. The settings you provided are ok and I tested them with WordPress-Core,WordPress-Docs and work as expected. Please check whether this issue occurs with vscode-phpcs 0.7.0

**Comment by mike131** [comment]
This may be related or I may be using incorrectly, but I have
`
{
  "phpcs.standard": "/path/to/phpcs-ruleset.xml"
}
`

I use a custom ruleset.

**Comment by ikappas** [comment]
@mike131 This is not related. But is the path/to/phpcs-ruleset.xml relative to the project?

**Comment by mike131** [comment]
@ikappas No,I am using the absolute path.

**Comment by ikappas** [comment]
@mike131 windows or mac?

**Comment by mike131** [comment]
@ikappas Mac

**Comment by ikappas** [comment]
@mike131 I can not reproduce your issue as setting an absolute path for phpcs.standard works as expected in my mac even outside of the project folder. ie /Users/mike131/phpcs-ruleset.xml. I even tried with a ruleset with spaces in the path but it still works as expected. I am closing this issue as it @mike131 issue is not relevant and I assume the main issue is resolved with 0.7.0.  Feel free to open a new issue if you continue to experience this.

**Comment by soniktrooth** [comment]
I'm having this same issue.

**Comment by andrebonon** [comment]
I'm having the same issue, and it is too annoying.

**Comment by ikappas** [comment]
@soniktrooth @andrebonon 
1. Which version of vscode-phpcs are you using?
2. Which version of phpcs are you using
3. How do you have phpcs installed (globally/composer)
3. What is the output of `phpcs  --report=json`

**Comment by soniktrooth** [comment]
@ikappas I think I've fixed it by uninstalling and reinstalling everything. Not sure what the issue was. Thanks.

**Comment by ikappas** [comment]
@soniktrooth @andrebonon Can you try the latest development branch to test whether your issues are resolved?

**Comment by itsam** [comment]
vs: 1.18.0
phpcs: 3.0.2 (stable)
installed via composer
the output of phpcs --report=json seems fine and as expected

Still I am having the same issue, namely "phpcs: Unexpected token P in JSON at position 0" in each file save.

Any ideas are welcome. Thanks!

**Comment by ikappas** [comment]
@itsam The extension on the marketplace is due for an update. Meanwhile, you can try the develop branch as it has a lot of fixes.

**Comment by itsam** [comment]
Not sure how to install outside marketplace. I followed the steps below but I cannot see phpcs on the installed extensions...

- install the Visual Studio Code npm extension
- clone this repository and checkout develop branch
- open the cloned repository folder using Visual Studio Code
- run VS Code task npm install <--- **not sure about that, just opened the terminal and typed npm install**

**Comment by ntwb** [comment]
I wonder if something like [cosmiconfig](https://www.npmjs.com/package/cosmiconfig) could help solve this 🤔

**Comment by ntwb** [comment]
Here's another example, I'd like to use the `WordPress` standard across multiple projects, as such I'd have this configured as:

```
{
  "phpcs.standard": "WordPress"
}
```

Then, each _WordPress_ project has it's own `phpcs.xml.dist` file for any customizations and tweaks, for example here's two WordPress Core that each use a unique `phpcs.xml.dist` file:
• https://core.trac.wordpress.org/browser/trunk/phpcs.xml.dist
• https://github.com/WordPress/gutenberg/blob/master/phpcs.xml.dist

So, it be great to be able to specific `WordPress` as my _default_ PHPCS _standard_ and to then be able to have a `phpcs.xml.dist` file read from a project folder to be used in conjunction, and then further, if a `phpcs.xml` exists to once again extend upon the previous `phpcs.xml.dist` and `WordPress` _standards_ previously loaded.

**Comment by rahulnever2far** [comment]
Multiple standards at the same time would be very helpful.

Personally, I'm looking for something like @ntwb suggested. A way to use project-specific ruleset in conjunction with default standard defined in the settings. Maybe, the extension could automatically look for a file named `phpcs.xml`. If it exists it should automatically include it.

**Comment by ntwb** [comment]
> Maybe, the extension could automatically look for a file named `phpcs.xml`. If it exists it should automatically include it.

https://www.npmjs.com/package/cosmiconfig should be able to handle merging and/or nested configs

**Comment by ikappas** [comment]
I've just pushed 1.0.0-beta.6 on the development branch adding automatic configuration search. This is the last feature for the 1.0.0 release which needs some testing before 1.0.0 launch.

**Comment by ikappas** [comment]
I just released vscode-phpcs 1.0.0 which supports the VSCode 1.20+ and includes many enhancements most notable of which are:

1. multi root support
2. lint as you type support
3. automatic configuration search support
4. ability to set phpcs.executablePath
5. ability to set phpcs.showWarnings
7. ability to set phpcs.showSources
8. ability to set phpcs.composerJsonPath to something other that workspaceRoot/composer.json
9. ability to enable phpcs.trace.server with messages or verbose option for tracing errors.

I would like to give my thanks to all the people that helped track down issues in the previous betas and hope the new features help you get things done easier.
On that note, I am closing this issue to properly track any new issues in the new release. If you find this issue persists or any other issue please feel free to open a new ticket for v.1.0.0.

**Comment by josephfusco** [comment]
I am also looking for something like @ntwb suggested. Defining "WordPress" along side with a WordPress plugin's own `.phpcs.xml.dist` however I am able to get it working.

My workspace settings:
```json
{
  "phpcs.standard": "WordPress",
  "phpcbf.standard": "WordPress",
  "phpcs.executablePath": "./vendor/bin/phpcs",
  "phpcs.composerJsonPath": "composer.json",
  "phpcs.autoConfigSearch": true,
  "phpcs.enable": true,
  "phpcs.errorSeverity": 5,
  "phpcs.ignorePatterns": [],
  "phpcs.showSources": true,
  "phpcs.showWarnings": true,
  "phpcs.trace.server": "verbose",
  "phpcs.warningSeverity": 5,
}
```

My understanding so far is that if I have `autoConfigSearch: true`, it will find the project's `.phpcs.xml.dist` and automatically load that on top of the default WordPress ruleset.
